### PR TITLE
tutorial.lisp: copy and paste commands

### PR DESCRIPTION
**Pull request by pdelfino**
Hi guys,

As discussed on issue #1577  and [here](https://github.com/atlas-engineer/nyxt/commit/2716a69bbf54eb38480de27cf6ccc08c181e37fd#commitcomment-52850352) I think the tutorial should 
describe the powerful features of copying and pasting provided by Nyxt.

There are 11 commands. I am aware that some (3) may be
conflated, that some (new) bindings may be created, and that some commands
might have their name changed. Even so, I thought that doing this PR
could be useful.

Either to trigger the mentioned changes or to help people understand
the current version. This content would have helped me 8 weeks ago
when I was new to Nyxt.

Basically, I used the docstrings from the command definition. The
exception was adding that `copy-system-information` could be useful
for bug reports and introducing the article `the` in some definitions
(Grammarly suggestion). The docstrings could incorporate these
changes.

Finally, I noticed an inconsistency on `copy` commands explanations. The
definitions use sometimes *copies* and sometimes *saves*. Maybe it would be better to use
just one verb (I prefer *save*).

### CODE REVIEW

**Comment by Ambrevar** [comment]
Looks good to me!

**Comment by edgar-vincent** [comment]
This looks very useful indeed! There were some commands I didn't know about!

On line 128, I'd remove `Starting from the classic`. I think it is a little informal and not particularly necessary.

Line 131: s/active-element/active element/
Line 138: s/user inputted/user-inputted
Line 143: s/suggestion/suggestions/

**Comment by pdelfino** [comment]
> This looks very useful indeed! There were some commands I didn't know about!

I am glad to hear that. You seem to be a Nyxt heavy user :)

> On line 128, I'd remove `Starting from the classic`. I think it is a little informal and not particularly necessary.

Ok. I am changing to just `Starting from`

On:
> Line 131: s/active-element/active element/
> Line 138: s/user inputted/user-inputted
> Line 143: s/suggestion/suggestions/

These words all come from docstrings. I just copied them. But the docstrings can be changed in the function definition.

I applied `command-docstring-first-sentence` recently created by @Ambrevar and I also changed the command definitions using @edgar-vincent 's suggestions.

File: source/tutorial.lisp [hunk 1]
```diff
@@ -123,6 +123,29 @@ full URL including the 'http://' prefix.")
...
+    (:li (command-markup 'nyxt/web-mode:copy) ": " (command-docstring-first-sentence 'nyxt/web-mode:copy))
+    (:li (command-markup 'nyxt/web-mode:paste) ": " (command-docstring-first-sentence 'nyxt/web-mode:paste)))
+   (:p "Passing through webpage's data:")
+   (:ul
+    (:li (command-markup 'copy-url) ": " (command-docstring-first-sentence 'copy-url))
+    (:li (command-markup 'copy-title) ": " (command-docstring-first-sentence 'copy-title))
+    (:li (command-markup 'nyxt/web-mode:copy-placeholder) ": " (command-docstring-first-sentence 'nyxt/web-mode:copy-placeholder))
+    (:li (command-markup 'nyxt/web-mode:copy-hint-url) ": " (command-docstring-first-sentence 'nyxt/web-mode:copy-hint-url)))
+   (:p "Leveraging password managers: ")
+   #+nil
```
Comment (pdelfino) [inline source/tutorial.lisp:139]: @Ambrevar ,

For some reason that I do not know, I cannot compile Nyxt if the expression bellow `#+nil` is valued.

Why is this happening? Is this related to the macro `command-docstring-first-sentence`?

File: source/tutorial.lisp [hunk 2]
```diff
@@ -123,6 +123,29 @@ full URL including the 'http://' prefix.")
...
+   (:ul 
+    (:li (command-markup 'copy-username) ": " (command-docstring-first-sentence 'copy-username))
+    (:li (command-markup 'copy-password) ": " (command-docstring-first-sentence 'copy-password))
+    (:li (command-markup 'copy-password-prompt-details) ": " (command-docstring-first-sentence 'copy-password-prompt-details)))
+   (:p "And more: ")
+   #+nil
```
Comment (pdelfino) [inline source/tutorial.lisp:145]: The same problem happens here. If I take out the `#+nil` nothing works.

How can I fix this?

File: source/tutorial.lisp (hunk 1 above)
Comment (Ambrevar) [reply source/tutorial.lisp:139]: You need to tell me more :)
What's the compile error?

Comment (pdelfino) [reply source/tutorial.lisp:139]: When loading the system, the compilation does not work as usual. It shows this message:

```lisp
; SLIME 2.26.1WARNING: System definition file #P"/home/pedro/.quicklisp/dists/quicklisp/software/calispel-20170830-git/calispel.asd" contains definition for system "calispel-test". Please only define "calispel" and secondary systems with a name starting with "calispel/" (e.g. "calispel/test") in that file.
WARNING: System definition file #P"/home/pedro/.quicklisp/dists/quicklisp/software/cl-json-20141217-git/cl-json.asd" contains definition for system "cl-json.test". Please only define "cl-json" and secondary systems with a name starting with "cl-json/" (e.g. "cl-json/test") in that file.
WARNING: System definition file #P"/home/pedro/.quicklisp/dists/quicklisp/software/cl-ppcre-20190521-git/cl-ppcre.asd" contains definition for system "cl-ppcre-test". Please only define "cl-ppcre" and secondary systems with a name starting with "cl-ppcre/" (e.g. "cl-ppcre/test") in that file.
WARNING: System definition file #P"/home/pedro/.quicklisp/dists/quicklisp/software/cl-ppcre-20190521-git/cl-ppcre-unicode.asd" contains definition for system "cl-ppcre-unicode-test". Please only define "cl-ppcre-unicode" and secondary systems with a name starting with "cl-ppcre-unicode/" (e.g. "cl-ppcre-unicode/test") in that file.
WARNING: System definition file #P"/home/pedro/.quicklisp/dists/quicklisp/software/s-xml-20150608-git/s-xml.asd" contains definition for system "s-xml.test". Please only define "s-xml" and secondary systems with a name starting with "s-xml/" (e.g. "s-xml/test") in that file.
WARNING: System definition file #P"/home/pedro/.quicklisp/dists/quicklisp/software/s-xml-20150608-git/s-xml.asd" contains definition for system "s-xml.examples". Please only define "s-xml" and secondary systems with a name starting with "s-xml/" (e.g. "s-xml/test") in that file.
WARNING: System definition file #P"/home/pedro/.quicklisp/dists/quicklisp/software/flexi-streams-20200925-git/flexi-streams.asd" contains definition for system "flexi-streams-test". Please only define "flexi-streams" and secondary systems with a name starting with "flexi-streams/" (e.g. "flexi-streams/test") in that file.
;;; Computing Hangul syllable names
WARNING: redefining S-SERIALIZATION::SERIALIZE-SEXP-INTERNAL in DEFGENERIC
WARNING: redefining CL-PREVALENCE:GET-ID in DEFGENERIC
WARNING: redefining CL-PREVALENCE:GET-NAME in DEFGENERIC
WARNING: redefining (COMMON-LISP:SETF CL-PREVALENCE:GET-NAME) in DEFGENERIC
WARNING: redefining CL-PREVALENCE:GET-SIZE in DEFGENERIC
WARNING: redefining CL-PREVALENCE:GET-MIME-TYPE in DEFGENERIC
WARNING: redefining (COMMON-LISP:SETF CL-PREVALENCE:GET-MIME-TYPE) in DEFGENERIC
WARNING: redefining CL-PREVALENCE:GET-KEYWORDS in DEFGENERIC
WARNING: redefining (COMMON-LISP:SETF CL-PREVALENCE:GET-KEYWORDS) in DEFGENERIC
WARNING: redefining LOG4CL-IMPL:APPENDER-ADDED in DEFGENERIC
WARNING: redefining LOG4CL-IMPL:APPENDER-REMOVED in DEFGENERIC
WARNING: redefining LOG4CL-IMPL::PROPERTY-ALIST in DEFGENERIC
WARNING: SWANK/BACKEND also exports the following symbols:
  (SWANK/BACKEND:SLDB-STEP-NEXT SWANK/BACKEND:CALL-WITH-LOCK-HELD
                                SWANK/BACKEND:WHO-BINDS
                                SWANK/BACKEND:MAKE-WEAK-KEY-HASH-TABLE
                                SWANK/BACKEND:FIND-DEFINITIONS
                                SWANK/BACKEND:FORMAT-STRING-EXPAND
                                SWANK/BACKEND:FRAME-CALL
                                SWANK/BACKEND:MAKE-AUTO-FLUSH-THREAD
                                SWANK/BACKEND:FIND-REGISTERED
                                SWANK/BACKEND:PACKAGE-LOCAL-NICKNAMES
                                SWANK/BACKEND:INITIALIZE-MULTIPROCESSING
                                SWANK/BACKEND:DESCRIBE-SYMBOL-FOR-EMACS
                                SWANK/BACKEND:CALL-WITH-DEBUGGER-HOOK
                                SWANK/BACKEND:SET-DEFAULT-INITIAL-BINDING
                                SWANK/BACKEND:WAKE-THREAD
                                SWANK/BACKEND:WHO-CALLS
                                SWANK/BACKEND:CALL-WITH-DEBUGGING-ENVIRONMENT
                                SWANK/BACKEND:COMPILER-MACROEXPAND-1
                                SWANK/BACKEND:TYPE-SPECIFIER-P
                                SWANK/BACKEND:DESCRIBE-PRIMITIVE-TYPE
                                SWANK/BACKEND:DISASSEMBLE-FRAME
                                SWANK/BACKEND:THREAD-ID SWANK/BACKEND:WRAP
                                SWANK/BACKEND:FLOAT-INFINITY-P
                                SWANK/BACKEND:SAVE-IMAGE
                                SWANK/BACKEND:WHO-REFERENCES
                                SWANK/BACKEND:PRINT-FRAME
                                SWANK/BACKEND:GRAY-PACKAGE-NAME
                                SWANK/BACKEND:CALL-WITH-SYNTAX-HOOKS
                                SWANK/BACKEND:GUESS-EXTERNAL-FORMAT
                                SWANK/BACKEND:SWANK-COMPILE-FILE
                                SWANK/BACKEND:LISP-IMPLEMENTATION-PROGRAM
                                SWANK/BACKEND:SOCKET-FD
                                SWANK/BACKEND:INSTALL-DEBUGGER-GLOBALLY
                                SWANK/BACKEND:BACKGROUND-SAVE-IMAGE
                                SWANK/BACKEND:VALID-FUNCTION-NAME-P
                                SWANK/BACKEND:WHO-MACROEXPANDS
                                SWANK/BACKEND:EMACS-CONNECTED
                                SWANK/BACKEND:SLDB-STEP-INTO
                                SWANK/BACKEND:FIND-SOURCE-LOCATION
                                SWANK/BACKEND:ACTIVATE-STEPPING
                                SWANK/BACKEND:WAIT-FOR-INPUT SWANK/BACKEND:DUP
                                SWANK/BACKEND:UNPROFILE SWANK/BACKEND:PROFILE
                                SWANK/BACKEND:SLDB-STEPPER-CONDITION-P
                                SWANK/BACKEND:WHO-SPECIALIZES
                                SWANK/BACKEND:FUNCTION-NAME
                                SWANK/BACKEND:SLDB-STEP-OUT
                                SWANK/BACKEND:INSTALL-SIGINT-HANDLER
                                SWANK/BACKEND:THREAD-STATUS
                                SWANK/BACKEND:MAKE-LOCK
                                SWANK/BACKEND:ALL-THREADS
                                SWANK/BACKEND:CLOSE-SOCKET
                                SWANK/BACKEND:WHO-SETS
                                SWANK/BACKEND:FIND-LOCALLY-NICKNAMED-PACKAGE
                                SWANK/BACKEND:COMPUTE-BACKTRACE
                                SWANK/BACKEND:THREAD-ATTRIBUTES
                                SWANK/BACKEND:RECEIVE SWANK/BACKEND:EXEC-IMAGE
                                SWANK/BACKEND:CREATE-SOCKET
                                SWANK/BACKEND:DESCRIBE-DEFINITION
                                SWANK/BACKEND:SWANK-COMPILE-STRING
                                SWANK/BACKEND:RECEIVE-IF SWANK/BACKEND:SEND
                                SWANK/BACKEND:KILL-THREAD
                                SWANK/BACKEND:CALLS-WHO
                                SWANK/BACKEND:FILENAME-TO-PATHNAME
                                SWANK/BACKEND:LIST-CALLEES
                                SWANK/BACKEND:ADD-SIGIO-HANDLER
                                SWANK/BACKEND:CALL-WITH-USER-BREAK-HANDLER
                                SWANK/BACKEND:DEFAULT-READTABLE-ALIST
                                SWANK/BACKEND:COMPILER-MACROEXPAND
                                SWANK/BACKEND:REGISTER-THREAD
                                SWANK/BACKEND:LOCAL-PORT
                                SWANK/BACKEND:FIND-EXTERNAL-FORMAT
                                SWANK/BACKEND:THREAD-ALIVE-P
                                SWANK/BACKEND:COLLECT-MACRO-FORMS
                                SWANK/BACKEND:SET-STREAM-TIMEOUT
                                SWANK/BACKEND:INTERRUPT-THREAD
                                SWANK/BACKEND:FRAME-PACKAGE
                                SWANK/BACKEND:CONDITION-EXTRAS
                                SWANK/BACKEND:UNWRAP
                                SWANK/BACKEND:FRAME-VAR-VALUE
                                SWANK/BACKEND:EVAL-IN-FRAME
                                SWANK/BACKEND:STRING-TO-UTF8
                                SWANK/BACKEND:PRINT-CONDITION
                                SWANK/BACKEND:LISP-IMPLEMENTATION-TYPE-NAME
                                SWANK/BACKEND:MAKE-WEAK-VALUE-HASH-TABLE
                                SWANK/BACKEND:ADD-FD-HANDLER
                                SWANK/BACKEND:UTF8-TO-STRING
                                SWANK/BACKEND:PREFERRED-COMMUNICATION-STYLE
                                SWANK/BACKEND:THREAD-NAME
                                SWANK/BACKEND:FRAME-RESTARTABLE-P
                                SWANK/BACKEND:WRAPPED-P
                                SWANK/BACKEND:FIND-THREAD
                                SWANK/BACKEND:MACROEXPAND-ALL
                                SWANK/BACKEND:LIST-CALLERS
                                SWANK/BACKEND:PATHNAME-TO-FILENAME
                                SWANK/BACKEND:MAKE-INPUT-STREAM
                                SWANK/BACKEND:REMOVE-FD-HANDLERS
                                SWANK/BACKEND:ARGLIST
                                SWANK/BACKEND:FRAME-LOCALS
                                SWANK/BACKEND:EVAL-CONTEXT
                                SWANK/BACKEND:RETURN-FROM-FRAME
                                SWANK/BACKEND:FLOAT-NAN-P
                                SWANK/BACKEND:MAKE-FD-STREAM
                                SWANK/BACKEND:MAKE-OUTPUT-STREAM
                                SWANK/BACKEND:TOGGLE-TRACE
                                SWANK/BACKEND:CALL-WITH-COMPILATION-HOOKS
                                SWANK/BACKEND:GETPID
                                SWANK/BACKEND:SLDB-BREAK-AT-START
                                SWANK/BACKEND:SPAWN
                                SWANK/BACKEND:HASH-TABLE-WEAKNESS
                                SWANK/BACKEND:CHARACTER-COMPLETION-SET
                                SWANK/BACKEND:ACCEPT-CONNECTION
                                SWANK/BACKEND:CURRENT-THREAD
                                SWANK/BACKEND:FRAME-CATCH-TAGS
                                SWANK/BACKEND:REMOVE-SIGIO-HANDLERS
                                SWANK/BACKEND:COMMAND-LINE-ARGS)
See also:
  The ANSI Standard, Macro DEFPACKAGE
  The SBCL Manual, Variable *ON-PACKAGE-VARIANCE*
WARNING: SWANK also exports the following symbols:
  (SWANK:TOGGLE-PROFILE-FDEFINITION SWANK:INSPECT-IN-FRAME
                                    SWANK:DEBUG-NTH-THREAD
                                    SWANK:FIND-SOURCE-LOCATION-FOR-EMACS
                                    SWANK:COMPILE-MULTIPLE-STRINGS-FOR-EMACS
                                    SWANK:DESCRIBE-DEFINITION-FOR-EMACS
                                    SWANK:INIT-INSPECTOR
                                    SWANK:SWANK-MACROEXPAND SWANK:LOAD-FILE
                                    SWANK:SLDB-BREAK-WITH-DEFAULT-DEBUGGER
                                    SWANK:TOGGLE-DEBUG-ON-SWANK-ERROR
                                    SWANK:PACKAGE= SWANK:SIMPLE-COMPLETIONS
                                    SWANK:SLDB-NEXT SWANK:COMPLETIONS
                                    SWANK:ARGLIST-DISPATCH
                                    SWANK:SDLB-PRINT-CONDITION
                                    SWANK:SWANK-MACROEXPAND-1
                                    SWANK:FIND-DEFINITION-FOR-THING
                                    SWANK:COMPILE-FILE-FOR-EMACS
                                    SWANK:DESCRIBE-SYMBOL SWANK:AUTODOC
                                    SWANK:LOOKUP-PRESENTED-OBJECT-OR-LOSE
                                    SWANK:EVAL-STRING-IN-FRAME
                                    SWANK:INSPECT-PRESENTATION SWANK:SLDB-OUT
                                    SWANK:VALUE-FOR-EDITING
                                    SWANK:INSPECT-FRAME-VAR
                                    SWANK:INTERACTIVE-EVAL-REGION
                                    SWANK:SLDB-DISASSEMBLE SWANK:SIMPLE-BREAK
                                    SWANK:SWANK-EXPAND
                                    SWANK:ASDF-DETERMINE-SYSTEM
                                    SWANK:CONNECTION-INFO
                                    SWANK:EXPORT-SYMBOL-FOR-EMACS
                                    SWANK:OPERATOR-ARGLIST
                                    SWANK:*FUZZY-DUPLICATE-SYMBOL-FILTER*
                                    SWANK:PROFILE-BY-SUBSTRING
                                    SWANK:LIST-ASDF-SYSTEMS
                                    SWANK:FLOW-CONTROL-TEST
                                    SWANK:FRAME-LOCALS-AND-CATCH-TAGS
                                    SWANK:INVOKE-NTH-RESTART
                                    SWANK:SWANK-EXPAND-1
                                    SWANK:EVAL-AND-GRAB-OUTPUT
                                    SWANK:FUZZY-COMPLETION-SELECTED
                                    SWANK:INSPECT-NTH-PART SWANK:XREFS
                                    SWANK:SWANK-DELETE-PACKAGE
                                    SWANK:UNDEFINE-FUNCTION
                                    SWANK:COMMIT-EDITED-VALUE
                                    SWANK:PPRINT-EVAL-STRING-IN-FRAME
                                    SWANK:INSPECTOR-NTH-PART
                                    SWANK:DESCRIBE-FUNCTION
                                    SWANK:EXPORT-STRUCTURE
                                    SWANK:THROW-TO-TOPLEVEL
                                    SWANK:INSPECTOR-REINSPECT SWANK:SLDB-ABORT
                                    SWANK:INIT-PRESENTATIONS
                                    SWANK:SWANK-PROFILE-PACKAGE
                                    SWANK:DEBUGGER-INFO-FOR-EMACS
                                    SWANK:COMPILE-FILE-IF-NEEDED
                                    SWANK:START-SWANK-SERVER-IN-THREAD
                                    SWANK:COMPLETE-FORM SWANK:INSPECTOR-HISTORY
                                    SWANK:INSPECT-CURRENT-CONDITION
                                    SWANK:WHO-DEPENDS-ON
                                    SWANK:QUIT-THREAD-BROWSER
                                    SWANK:SWANK-TOGGLE-TRACE SWANK:PPRINT-EVAL
                                    SWANK:FRAME-PACKAGE-NAME
                                    SWANK:COMPILE-STRING-FOR-EMACS
                                    SWANK:INVOKE-NTH-RESTART-FOR-EMACS
                                    SWANK:FUZZY-COMPLETIONS
                                    SWANK:RE-EVALUATE-DEFVAR
                                    SWANK:QUIT-INSPECTOR
                                    SWANK:DELETE-SYSTEM-FASLS
                                    SWANK:FIND-DEFINITIONS-FOR-EMACS
                                    SWANK:SWANK-MACROEXPAND-ALL
                                    SWANK:RELOAD-SYSTEM SWANK:KILL-NTH-THREAD
                                    SWANK:XREF SWANK:INSPECTOR-RANGE
                                    SWANK:PPRINT-INSPECTOR-PART
                                    SWANK:SET-PACKAGE
                                    SWANK:ASDF-SYSTEM-DIRECTORY
                                    SWANK:APROPOS-LIST-FOR-EMACS
                                    SWANK:LOOKUP-AND-SAVE-PRESENTED-OBJECT-OR-LOSE
                                    SWANK:ASDF-SYSTEM-FILES
                                    SWANK:LIST-ALL-SYSTEMS-KNOWN-TO-ASDF
                                    SWANK:LIST-ALL-SYSTEMS-IN-CENTRAL-REGISTRY
                                    SWANK:DISASSEMBLE-FORM SWANK:MOP
                                    SWANK:SLDB-CONTINUE SWANK:INSPECTOR-POP
                                    SWANK:LOOKUP-PRESENTED-OBJECT
                                    SWANK:LIST-THREADS SWANK:DESCRIBE-INSPECTEE
                                    SWANK:IO-SPEED-TEST
                                    SWANK:SWANK-FORMAT-STRING-EXPAND
                                    SWANK:OPERATE-ON-SYSTEM-FOR-EMACS
                                    SWANK:UPDATE-INDENTATION-INFORMATION
                                    SWANK:COMPLETIONS-FOR-KEYWORD
                                    SWANK:COMPLETIONS-FOR-CHARACTER
                                    SWANK:SWANK-REQUIRE
                                    SWANK:UNEXPORT-SYMBOL-FOR-EMACS
                                    SWANK:INSPECTOR-NEXT
                                    SWANK:ASDF-SYSTEM-LOADED-P
                                    SWANK:UNINTERN-SYMBOL SWANK:PING
                                    SWANK:SWANK-COMPILER-MACROEXPAND
                                    SWANK:INTERACTIVE-EVAL
                                    SWANK:SWANK-COMPILER-MACROEXPAND-1
                                    SWANK:UNTRACE-ALL
                                    SWANK:LIST-ALL-PACKAGE-NAMES
                                    SWANK:CLEAR-REPL-RESULTS
                                    SWANK:TOGGLE-BREAK-ON-SIGNALS
                                    SWANK:INSPECTOR-CALL-NTH-ACTION
                                    SWANK:BACKTRACE SWANK:INSPECTOR-EVAL
                                    SWANK:DOCUMENTATION-SYMBOL
                                    SWANK:INSPECTOR-TOGGLE-VERBOSE
                                    SWANK:SLDB-RETURN-FROM-FRAME)
See also:
  The ANSI Standard, Macro DEFPACKAGE
  The SBCL Manual, Variable *ON-PACKAGE-VARIANCE*
WARNING:
   redefining EMACS-INSPECT (#<SB-PCL:SYSTEM-CLASS COMMON-LISP:T>) in DEFMETHOD
WARNING:
   redefining EMACS-INSPECT (#<SB-PCL:SYSTEM-CLASS COMMON-LISP:T>) in DEFMETHOD
WARNING:
   redefining THREAD-FOR-EVALUATION (#<STRUCTURE-CLASS SWANK::MULTITHREADED-CONNECTION>
                                     #<SB-MOP:EQL-SPECIALIZER :FIND-EXISTING>) in DEFMETHOD
WARNING: redefining SWANK::SYMBOL-INDENTATION in DEFUN
WARNING: redefining SWANK::MACRO-INDENTATION in DEFUN
WARNING: redefining HU.DWIM.DEFCLASS-STAR::PROCESS-SLOT-DEFINITION in DEFUN
STYLE-WARNING:
   Generic function FSET:ITERATOR clobbers an earlier FTYPE proclamation
   (FUNCTION (T &KEY &ALLOW-OTHER-KEYS) (VALUES FUNCTION &REST T)) for the same
   name with (FUNCTION (T &KEY &ALLOW-OTHER-KEYS) *).

CL-USER> 
```

Comment (pdelfino) [reply source/tutorial.lisp:139]: If try to be inside the package:

```lisp

CL-USER> (in-package :nyxt)

```

The REPL keeps going on forever to execute it and never finishes it.

Comment (pdelfino) [reply source/tutorial.lisp:139]: If I try to run nyxt being in `CL-USER` the same problem also happens:
```lisp
CL-USER> (nyxt:start)
```

**Comment by jmercouris** [comment]
Closing this PR since these commits have now been merged.
